wikimedia/wikipedia-ios
# Submit Payment - update error display

**Phabricator:**
https://phabricator.wikimedia.org/T345846

### Notes
When testing a 500 HTTP status, I noticed some of the generic error displays were not helpful (see screenshots). This PR updates any errors that will be displayed alongside a generic message to display the named enum. My hope is that this will help us zero in on issues if we see user complaints.

### Test Steps
1. Add this code around [this line](https://github.com/wikimedia/wikipedia-ios/blob/main/WKData/Sources/WKData/Service/WKBasicService.swift#L96): 

```
        completion(nil, nil, WKServiceError.invalidHttpResponse(500))
        return
```

2. Log in as Apple Pay sandbox user in iOS Settings, add test Apple Pay card in Wallet, update device region to NL or IT, change device date to Oct 17th. Launch app on this branch, background/foreground and pull to refresh to load campaigns (whew). 
3. Try submitting a payment on the Apple Pay form from Settings. Confirm you see the 3rd screenshot.

### Screenshots/Videos

Before:
![Settings 4](https://github.com/wikimedia/wikipedia-ios/assets/3620196/7516b3d0-b60c-4d46-8bf1-b4491a279a84)
![IMG_0249](https://github.com/wikimedia/wikipedia-ios/assets/3620196/35302972-27c9-47e5-a2dd-ec9575542adc)


After:
![IMG_0250](https://github.com/wikimedia/wikipedia-ios/assets/3620196/bcbeccd9-9f05-4460-8537-29eed0d2411f)
![IMG_0251](https://github.com/wikimedia/wikipedia-ios/assets/3620196/98333df9-5449-4848-ade2-28dea2a5fa50)
